src/functional/ffront/foast_to_itir.py

```diff
@@ -160,93 +59,93 @@ def visit_FieldOperator(self, node: foast.FieldOperator, **kwargs) -> itir.Funct
+    ) -> itir.FunCall:
+        *assigns, return_stmt = body
+        current_expr = self.visit(return_stmt, **kwargs)
+
+        for assign in assigns[-1::-1]:
+            current_expr = mi.let(*self.visit(assign, **kwargs))(current_expr)
+
+        return mi.deref(current_expr)
+
+    def _make_lambda_param_names(self, node: foast.Expr, **kwargs) -> list[str]:
+        def is_field(expr: foast.Expr) -> bool:
+            return TypeInfo(expr.type).is_field_type
+
+        names = list(
+            node.iter_tree().if_isinstance(foast.Name).filter(is_field).getattr("id").unique()
+        )
+        return [name for name in names]
+
+    def visit_Assign(self, node: foast.Assign, **kwargs) -> tuple[itir.Sym, itir.Expr]:
+        sym = self.visit(node.target, **kwargs)
+        expr = self.visit(node.value, **kwargs)
+        return sym, expr
 
     def visit_Return(self, node: foast.Return, **kwargs) -> itir.Expr:
-        result = self.visit(node.value, **kwargs)
-        return result
-
-    def visit_Symbol(
-        self, node: foast.Symbol, *, symtable: dict[str, foast.Symbol], **kwargs
-    ) -> itir.Sym:
-        if not isinstance(node.type, common_types.FieldType):
-            raise NotImplementedError("Only Field symbols are supported right now.")
-        return itir.Sym(id=node.id)
-
-    def visit_Name(
-        self,
-        node: foast.Name,
-        *,
-        symtable: dict[str, foast.Symbol],
-        shift_args: Optional[list[Union[itir.OffsetLiteral, itir.IntLiteral]]] = None,
-        **kwargs,
-    ) -> Union[itir.SymRef, itir.FunCall]:
-        # always shift a single field, not a field expression
-        if _name_is_field(node) and shift_args:
-            result = ItirDerefFactory(
-                args__0=ItirShiftFactory(shift_args=shift_args, args__0=itir.SymRef(id=node.id))
-            )
-            return result
-        return ItirDerefFactory(args__0=itir.SymRef(id=node.id))
+        return self.visit(node.value)
+
+    def visit_Symbol(self, node: foast.Symbol, **kwargs) -> itir.Sym:
+        return mi.sym(node.id)
+
+    def visit_Name(self, node: foast.Name, **kwargs) -> itir.SymRef:
+        return mi.ref(node.id)
 
     def visit_Subscript(self, node: foast.Subscript, **kwargs) -> itir.FunCall:
-        return ItirFunCallFactory(
-            name="tuple_get",
-            args=[self.visit(node.value, **kwargs), itir.IntLiteral(value=node.index)],
-        )
+        param_names = self._make_lambda_param_names(node, **kwargs)
+        return mi.lift(
+            mi.lambda_(*param_names)(
+                mi.tuple_get(mi.deref(self.visit(node.value, **kwargs)), node.index)
+            )
+        )(*param_names)
 
     def visit_TupleExpr(self, node: foast.TupleExpr, **kwargs) -> itir.FunCall:
-        return ItirFunCallFactory(name="make_tuple", args=self.visit(node.elts, **kwargs))
+        args = [mi.deref(arg) for arg in self.visit(node.elts, **kwargs)]
```

Comment:
This means there is currently no way of returning two fields from a field operator by our chosen path of creating a stencil and then lifting it, since

```python
return field1, field2
```

becomes
```python
return deref(make_tuple(iter1, iter2))
```

Which is invalid.